mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceCluster.java

```diff
@@ -44,6 +44,8 @@ public interface ResourceCluster extends ResourceClusterGateway {
      */
     String getName();
 
+    CompletableFuture<Ack> downloadArtifactsOnTaskExecutor(TaskExecutorID taskExecutorID);
```

Comment:
What is the semantics you are thinking of for the caching story? I was surprised to find out about this endpoint on the RC.

Reply:
My idea is to persist the mapping RC -> artifacts and get it when TEs get registered. In the future we can probably evolve the structure and make it more sophisticated. But for now this mapping I guess is fine.